jax/interpreters/masking.py

```diff
@@ -369,7 +368,7 @@ def aval(self):
 
   @property
   def dtype(self):
-    return self.val.dtype
+    return self.val.dtype if hasattr(self.val, 'dtype') else type(self.val)
```

Comment:
@j-towns @mattjj Is there a better way to do this?

Reply:
iiuc `getattr(self.val, 'dtype', type(self.val))` should be semantically equivalent.

Reply:
That's better.